gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java

```diff
@@ -266,16 +266,45 @@ private Gradebook getGradebook(String siteId) {
 	}
 	
 	/**
-	 * Get a list of assignments in the gradebook in the specified site that the current user is allowed to access, sorted by sort order
-	 * 
+	 * Get a list of assignments in the gradebook in the specified site that the current user is allowed to access, sorted by default sort order
+	 *
 	 * @param siteId the siteId
 	 * @return a list of assignments or null if no gradebook
 	 */
 	public List<Assignment> getGradebookAssignments(String siteId) {
+		return getGradebookAssignments(siteId, false);
+	}
+
+	/**
+	 * Get a list of assignments in the gradebook in the specified site that the current user is allowed to access, sorted by sort order or by categorised sort order
+	 *
+	 * @param sortByCategory whether to pre-sort the assignments based on their categorized sort order
+	 * @return a list of assignments or null if no gradebook
+	 */
+	public List<Assignment> getGradebookAssignments(boolean sortByCategory) {
```

Comment:
It seemed like a good idea to move the pre-sorting to the service and I overloaded `getGradebookAssignments` to support a new sort parameter as the `GradebookUISettings` is currently limited to the page.  Alternatively I can move knowledge of `GradebookUISettings' to the service.. which seems less dirty?